src/raven.js

```diff
@@ -777,11 +777,16 @@ Raven.prototype = {
             var xhrproto = XMLHttpRequest.prototype;
             fill(xhrproto, 'open', function(origOpen) {
                 return function (method, url) { // preserve arity
-                    this.__raven_xhr = {
-                        method: method,
-                        url: url,
-                        status_code: null
-                    };
+
+                    // intentionally checking against 0 here (start of string)
+                    if (url.indexOf(self._globalEndpoint) !== 0) {
```

Comment:
I mean, the same argument can be said about the checking for the presence of `_globalEndpoint` in a request bound for their servers.

If we want to be precise, we should validate the actual hostname, etc. It'd be marginally more expensive since doing this validation on every XHR request. So tradeoffs.

imo it's unlikely for someone to be kicking off requests to their own servers with the public key in the url, but it's your call.